docs/design/data_plane_manager.adoc

```diff
@@ -0,0 +1,1189 @@
+
+. When message body is greater than 4k , the performance would drop greatly
+image:images/dpm-mq2.png[image,width=400,height=240]
+
+[Figure 7] Message size and the performance drop
+
+. Here is the Neutron way to start ovs and do processing
+image:images/neutron-dpm.png[image,width=660,height=384]
+
+[Figure 8] Neutron way to start ovs and do processing
+
+. GRPC is not fit for transfering big file since it basically focus more on the assumption of small numbers in the data
+
+== Database Schema
+
+[cols=",,,",options="header",]
+|===
+|Vpc_states |Subnet_ states |Port_ states |security_group_ states
+| | | |
+|===
+
+State could contains more same type data as array
+
+* DataPlaneManager would deploy the goal state configuration in batch to the ACA grouply in parallel then wait for the response and do next accordingly.
+
+== REST APIs
+
+=== API Snapshot
+
+[width="100%",cols="22%,12%,50%,17%"]
+|===
+|*API Name* |*Method* |*Request*|*Response*
+
+|Get GoalState
+|GET
+|/goalstate/{goalstateid}
+|GoalState Info
+<<Get_GoalState,[sample]>>
+
+|Create Port by GoalState
+|POST
+|/goalstate/{gsinfo}
+|Goal State list
+<<Create_Port_GoalState,[sample]>>
+
+|Delete Port by GoalState
+|DELETE
+|/goalstate/{gsinfo}
+|GOAL State
+<<Delete_Port_GoalState,[sample]>>
+
+|Update GoalState
+|PUT
+|/goalstate/{goalstates}
+|GOAL State
+<<Update_GoalState,[sample]>>
+
+|Bulk Create Port by GoalState
+|POST
+|/goalstate/{gsinfo}
+|Goal State list
+<<BULK_Create_Port_GoalState,[sample]>>
+
+|Bulk Delete Port by GoalState
+|DELETE
+|/goalstate/{gsinfo}
+|GOAL State
+<<BULK_Delete_Port_GoalState,[sample]>>
+
+|Create Vpc by GoalState
+|POST
+|/goalstate/{gsinfo}
+|Goal State list
+<<Create_Vpc_GoalState,[sample]>>
+
+|Delete Vpc by GoalState
+|DELETE
+|/goalstate/{gsinfo}
+|GOAL State
+<<Delete_Vpc_GoalState,[sample]>>
+
+|Bulk Create Vpc by GoalState
+|POST
+|/goalstate/{gsinfo}
+|Goal State list
+<<BULK_Create_Vpc_GoalState,[sample]>>
+
+|Bulk Delete Vpc by GoalState
+|DELETE
+|/goalstate/{gsinfo}
+|GOAL State
+<<BULK_Delete_Vpc_GoalState,[sample]>>
+
+|===
+
+=== API Specification
+
+* Get GoalState
+* Update GoalState
+* Create Port with GoalState
+* Delete Port with GoalState
+* Bulk Create Port with GoalState
+* Bulk Delete Port with GoalState
+* Create Vpc with GoalState
+* Delete Vpc with GoalState
+* Bulk Create Vpc with GoalState
+* Bulk Delete Vpc with GoalState
+
+anchor:Get_GoalState[]
+==== (1) Get GoalState Info By GoalState Request
+
+* {blank}
++
+____
+Method: GET
+____
+* {blank}
++
+____
+Request: 
+____
+
+____
+"/goalstate/{goalstateid}","/v4/goalstate/{goalstateid}"
+____
+
+* {blank}
++
+____
+Request Parameter: 
+____
+** @ PathVariable String portid
+* {blank}
++
+____
+Response: List of GoalState Information
+____
+* {blank}
++
+____
+Example
+____
+
+Request:
+
+http://serverIP:8080/goalstate/f37810eb-7f83-45fa-a4d4-1b31e75399df
+
+Response:
+
+[source,c++]
+------------------------------------------------------------
+{
```

Comment:
As discussed last week, DPM needs to understand what need to send down to hosts based on AlcorSchema , and provide APIs accordingly. Could you update the PR or raise questions if any? Thanks.